py-polars/docs/source/conf.py

```diff
@@ -182,3 +183,30 @@ def linkcode_resolve(domain, info):
     return (
         f"https://github.com/pola-rs/polars/blob/master/py-polars/polars/{fn}{linespec}"
     )
+
+
+def _minify_classpaths(s: str) -> str:
+    # strip private polars classpaths, leaving the classname:
+    # * "pli.Expr" -> "Expr"
+    # * "polars.internals.expr.expr.Expr" -> "Expr"
+    # * "polars.internals.lazyframe.frame.LazyFrame" -> "LazyFrame"
+    return re.sub(
+        pattern=r"~?(pli|polars\.(?:internals|datatypes)[\w.]+?)\.([A-Z][\w.]+)",
+        repl=r"\2",
+        string=s,
+    )
+
+
+def process_signature(app, what, name, obj, options, signature, return_annotation):
+    return (
+        _minify_classpaths(signature) if signature else signature,
+        _minify_classpaths(return_annotation)
+        if return_annotation
+        else return_annotation,
```

Comment:
Personally I don't really like this syntax - that alternative `if/else` syntax can be nice, but only for one-liners in my opinion. But that's just a style preference, I guess.

Reply:
I'm not a fan either - it _was_ a one-liner when I wrote it, hah. Must have hit the line-length reformatting limit for `black`... I'll tweak the variable name.

Reply:
Cleaned-up:
```python
def process_signature(app, what, name, obj, opts, sig, ret):
    return (
        _minify_classpaths(sig) if sig else sig,
        _minify_classpaths(ret) if ret else ret,
    )
```